[stdlib] Set PYTHONIOENCODING when building the stdlib

bnbarham opened the pull request:
Ubuntu 18.04 seems to be defaulting to 'ascii', causing an error when using line-directive combined with the utf-8 diagnostic output used when early swift syntax is enabled.

Also fix up a validation-test while I'm here.

bnbarham committed:
[stdlib] Set PYTHONIOENCODING when building the stdlib

Ubuntu 18.04 seems to be defaulting to 'ascii', causing an error when
using line-directive combined with the utf-8 diagnostic output used when
early swift syntax is enabled.

bnbarham committed:
[validation] Fix line-directive validation test

This was disabled due to:
```
Unable to decode the command from the command line:
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 13-20: surrogates not allowed
```

Note that this error isn't in `line-directive`, but in Python decoding
surrogates from the command line. Not sure why that's not allowed, but
printing from within a script works so change to that.

We already set `PYTHONIOENCODING=UTF8` in the environment in
`test/lit.cfg`, which makes sure stdout is UTF8 as opposed to ASCII.

Resolves rdar://92613094.

bnbarham commented:
@swift-ci Please Build Toolchain Ubuntu 18.04

bnbarham commented:
@swift-ci please test

etcwilde reviewed (approved):
Nice. Hope that works.

bnbarham committed:
[Test] Remove unnecessary PYTHONIOENCODING

PYTHONIOENCODING is already specified in `lit.cfg`. The change that
added this also added the `<`, which is what actually made this pass -
PYTHONIOENCODING only overrides the encoding of stdin/stdout/stderr.

bnbarham commented:
@xymus for the last commit

xymus reviewed (approved):
We may want to set `LC_ALL` and others instead, but setting this env var worked fine for the test.

bnbarham commented:
@swift-ci please smoke test Linux platform
